src/ui/public/field_format_editor/pattern/pattern.js

```diff
@@ -15,6 +15,10 @@ uiModules
 
       $scope.$bind('inputs', attrs.inputs);
       $scope.$bind('placeholder', attrs.placeholder);
+      // The id attribute is not always an angular expression but nearly always
+      // just a string passed in
+      $scope.id = attrs.id;
+      attrs.$observe('id', () => $scope.id = attrs.id);
```

Comment:
I think this directive needs `replace: true`, or else the directive shows up as a DOM element with an `id` on it, which I think will take precedence over the input's `id`.

Also, I think you can define this a little more simply via the `scope` property:

```javascript
scope: {
  id: '@',
},
```

Did you already give that a shot?